### PR TITLE
feat: Add Go Getting started guide

### LINKED ISSUE #208: [DOC] Getting started tutorial for Go
### Change in the documentation

Create a getting started tutorial for Go. The [JavaScript tutorial](https://docs.openfeature.dev/docs/tutorials/getting-started/node) can be used as a reference.

### PR DESCRIPTION
**Pull request by Kavindu-Dodan**
## This PR
Introduce Go getting started guide. 

Fixes #208

### CODE REVIEW

**Review by beeme1mr — APPROVED** [review]

File: docs/tutorials/getting-started/go.mdx [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+- You have Docker installed and running on the host system
+
+### Step 1: Create a new Go project
+
+To get started, let's create a new folder and initialize the project. This can be done by running following commands.
+
+```shell
+mkdir GoStart
+cd GoStart
+go mod init GoStart
```
Comment (skyerus) [inline docs/tutorials/getting-started/go.mdx:39]: I've not seen a go module include capitalised letters before, could we use lowercase?

File: docs/tutorials/getting-started/go.mdx [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+	})
+
+	engine.Run()
+}
+```
+
+To add Go Gin dependency to the application, run the following commands.
+
+```shell
+get github.com/gin-gonic/gin
```
Comment (skyerus) [inline docs/tutorials/getting-started/go.mdx:76]:
```suggestion
go get github.com/gin-gonic/gin
```

File: docs/tutorials/getting-started/go.mdx [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        // Initialize Go Gin
+        engine := gin.Default()
+
+        // Setup a simple endpoint
+        engine.GET("/hello", func(c *gin.Context) {
+-               c.JSON(http.StatusOK, defaultMessage)
+-               return
+
++               // Evaluate welcome-message feature flag
++               welcomeMessage, _ := client.BooleanValue(
```
Comment (skyerus) [inline docs/tutorials/getting-started/go.mdx:110]: I think checking the error is worthwhile here for the sake of demonstrating good practice

File: docs/tutorials/getting-started/go.mdx [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+Now you can visit the url [http://localhost:8080/hello](http://localhost:8080/hello) and observe the message **Hello!**.
+
+<WhyDefaultContent/>
+
+### Step 5: Configure a provider (flagd)
+
+<FlagdContent/>
+
+Now let's make the required code changes in our application.
```
Comment (skyerus) [inline docs/tutorials/getting-started/go.mdx:149]:
```suggestion
Now, let's make the required code changes in our application.
```

File: docs/tutorials/getting-started/go.mdx [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+```
+
+Then, let's add `flagd provider` dependency with following commands.
+
+```shell
+go get github.com/open-feature/go-sdk-contrib/providers/flagd
+go mod tidy
+```
+
+Final code of the `main.go` file is given below,
```
Comment (skyerus) [inline docs/tutorials/getting-started/go.mdx:175]:
```suggestion
The complete `main.go` file is given below:
```

**Review by skyerus — COMMENTED** [review]
Nice :)

File: docs/tutorials/getting-started/go.mdx (hunk 3 above)
Comment (Kavindu-Dodan) [reply docs/tutorials/getting-started/go.mdx:110]: Initially, I had error handling and returned the default value. However, IMO this adds a little bit of complexity to the logic. I do agree with the point on showing good practices, but on the other hand, it is up to the developer who should be familiar with Go to improve the logic.

**Pull request merged**